src/erlazure_insights.erl

```diff
@@ -72,11 +72,22 @@ request_api(Url, Config) ->
 http_request(SubscriptionId, Filter, Select, Config) ->
     URL = "https://management.azure.com/subscriptions/" ++
     SubscriptionId ++
-    "/providers/microsoft.insights/eventtypes/management/values?" ++ "api-version=2014-04-01" ++ 
-    "&%24filter=" ++ Filter ++
-    "&%24select=" ++ Select,
+    "/providers/microsoft.insights/eventtypes/management/values?" ++
+    make_query_string([
+            {"api-version", "2014-04-01"},
+            {"%24filter", http_uri:encode(Filter)},
+            {"%24select", http_uri:encode(Select)}
+        ]),
     request_api(URL, Config).
 
+make_query_string(Params) ->
```

Comment:
this one can be experessed more clear: https://github.com/alertlogic/erlazure/blob/master/src/erlazure_ad_auth.erl#L91

can you please use this function of erlazure_ad_auth.erl, and also let's move this function to be in `erlazure_http.erl` instead of in two places